src/test/java/org/skife/jdbi/v2/BeanMapperTest.java

```
@@ -0,0 +1,270 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.skife.jdbi.v2;
+
+import org.easymock.EasyMockRunner;
+import org.easymock.Mock;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.skife.jdbi.v2.util.BigDecimalColumnMapper;
+import org.skife.jdbi.v2.util.IntegerColumnMapper;
+import org.skife.jdbi.v2.util.LongColumnMapper;
+import org.skife.jdbi.v2.util.StringColumnMapper;
+
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+
+@RunWith(EasyMockRunner.class)
+public class BeanMapperTest {
+
+    @Mock
+    ResultSet resultSet;
+
+    @Mock
+    ResultSetMetaData resultSetMetaData;
+
+    @Mock
+    StatementContext ctx;
+
+    BeanMapper<SampleBean> mapper = new BeanMapper<SampleBean>(SampleBean.class);
+
+    @Test
+    public void shouldSetValueOnPublicSetter() throws Exception {
+        expect(resultSetMetaData.getColumnCount()).andReturn(1).anyTimes();
+        expect(resultSetMetaData.getColumnLabel(1)).andReturn("longField");
+        replay(resultSetMetaData);
+
+        expect(ctx.columnMapperFor(Long.class)).andReturn(LongColumnMapper.WRAPPER);
+        replay(ctx);
+
+        expect(resultSet.getMetaData()).andReturn(resultSetMetaData);
+        Long aLongVal = 100l;
+        expect(resultSet.getLong(1)).andReturn(aLongVal);
+        expect(resultSet.wasNull()).andReturn(false);
+        replay(resultSet);
+
+        SampleBean sampleBean = mapper.map(0, resultSet, ctx);
+
+        assertSame(aLongVal, sampleBean.getLongField());
+    }
+
+    @Test
+    public void shouldHandleEmptyResult() throws Exception {
```

I'm a little confused about this behavior.  Seems to me if you attempt to map a Bean but do not include the appropriate columns in your `SELECT` it may be better to throw an error rather than silently return an empty bean.  Then again maybe selecting a subset of properties and defaulting the rest is useful.